packages/rum-core/src/performance-monitoring/metrics/inp/process.js

```diff
@@ -0,0 +1,256 @@
+   */
+  if (!isPerfInteractionCountSupported()) {
+    recorder.start(FIRST_INPUT)
+  }
+}
+
+/*
+   Processes the observed user interactions for INP purposes
+ */
+export function processUserInteractions(list) {
+  const entries = list.getEntries()
+  entries.forEach(entry => {
+    // Only entries with interactionId should be considered to calculate INP
+    if (!entry.interactionId) {
+      return
+    }
+
+    // All reported entries should be taken into account to estimate the total number of different interactions
+    updateInteractionCount(entry)
+
+    /**
+     * The duration can be lower than the recommended threshold because
+     * the observer will report events with durations >= 16 until browsers expose `performance.interactionCount`
+     * and the reason we don't store it is because the default Google's approach is to report as 0
+     * the INP if the slower duration is less than 40
+     * which is well below the recommended threshold.
+     */
+    if (entry.duration < INP_THRESHOLD) {
+      return
+    }
+
+    storeUserInteraction(entry)
+  })
+}
+
+export function calculateInp() {
+  if (inpState.longestInteractions.length === 0) {
+    /**
+     * The PerformanceObserver doesn't report all interactions because of the durationThreshold limit.
+     * Despite so, if the interactionCount is greater than 0, we should report 0
+     * which is the number/placeholder that Google has chosen as a way to convey that users had a good user experience.
+     */
+    if (interactionCount() > 0) {
+      return 0
+    }
+
+    // This means not activity at all, so nothing to report
+    return
+  }
+
+  /**
+   * If there are less than 50 interactions we will report the slowest interaction
+   * For 50 interactions or more, we will report the 98th percentile, which means picking the next in "line".
+   *
+   * One of the reasons why Google advocates for this is because it wouldn't be fair if your site would be penalized (SEO related),
+   * just because it received more interactions for an arbitrary reason.
+   */
+  const interactionIndex = Math.min(
+    inpState.longestInteractions.length - 1,
+    Math.floor(interactionCount() / 50)
+  )
+  const inp = inpState.longestInteractions[interactionIndex].duration
+
+  return inp
+}
+
+export function interactionCount() {
+  return performance.interactionCount || inpState.interactionCount
+}
+
+export function inpCandidates() {
+  return inpState.longestInteractions
+}
+
+export function setInteractionCount(count) {
+  inpState.interactionCount = count
+}
+
+export function restoreINPState() {
+  inpState.minInteractionId = Infinity
+  inpState.maxInteractionId = 0
+  inpState.interactionCount = 0
+  inpState.longestInteractions = []
+}
+
+// Stores user interaction
+function storeUserInteraction(entry) {
+  // Ignore interactions that are faster (or equal) than the least slow known interaction
+  const leastSlow =
```

Comment:
this is just an "optimization" (maybe a bit premature since the data structure cannot grow so much...). I would be open to delete this code if needed.

What it does:

if the newly captured entry is quicker than the least slow one then don't store it. The slowest one is the position 0 of the array, the least slowest one is in the last position (because of the sorting)

Reply:
thought again about this, I'm going to remove it, we are talking just about a few entries (10~). Let's write optimizations only when they are needed. "Nuke" mode activated 😄

Reply:
I added two additional tests to make sure this knowledge if represented in code 🚀

Reply:
I was going for the wording, The logic itself feels good to me. But thanks for adding the extra tests